sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -258,6 +300,29 @@ def decode_from_stream(self, in_stream, nested):
     return self._construct_from_components(
         [c.decode_from_stream(in_stream, True) for c in self._coder_impls])
 
+  def estimate_size(self, value):
+    """Estimates the encoded size of the given value, in bytes."""
+    estimated_size, unused_observables = (
+        self.get_estimated_size_and_observables(value))
+    return estimated_size
+
+  def get_estimated_size_and_observables(self, value):
+    """Returns estimated size of value along with any nested observables."""
+    values = self._extract_components(value)
+    estimated_size = 0
+    observables = []
+    for i in range(0, len(self._coder_impls)):
```

Comment:
I am keeping this form to be consistent with encode_to_stream(): we explicitly index into self._coder_impls here to make the Cython type cast for "c = self._coder_impls[i]" more obvious.